File: app/src/main/java/com/iamkatrechko/clipboardmanager/dialogs/DialogSetOrderType.java
package com.iamkatrechko.clipboardmanager.dialogs;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;

import com.iamkatrechko.clipboardmanager.R;
import com.iamkatrechko.clipboardmanager.util.UtilPreferences;

public class DialogSetOrderType extends DialogFragment {

    public static DialogSetOrderType newInstance() {
        return new DialogSetOrderType();
    }

    private void sendResult(String orderType) {
        if (getTargetFragment() == null) {
            return;
        }
        Intent a = new Intent();
        a.putExtra("orderType", orderType);
        getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, a);
    }

    @NonNull
    public Dialog onCreateDialog(Bundle bundle) {
        final String[] mItemsNames = getResources().getStringArray(R.array.orders_title);
        final String[] mItemsValues = getResources().getStringArray(R.array.orders_values);
        int currentOrderType = Integer.valueOf(UtilPreferences.getOrderType(getActivity()));

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("Сортировка:");
        builder.setSingleChoiceItems(mItemsNames, currentOrderType - 1, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int item) {
                sendResult(mItemsValues[item]);
                dismiss();
            }
        });
        return builder.create();
    }
}

File: app/src/main/java/com/iamkatrechko/clipboardmanager/view/dialog/DialogSetOrderType.kt
package com.iamkatrechko.clipboardmanager.view.dialog

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import com.iamkatrechko.clipboardmanager.R
import com.iamkatrechko.clipboardmanager.domain.param.values.OrderType
import com.iamkatrechko.clipboardmanager.domain.util.PrefsManager
import com.iamkatrechko.clipboardmanager.domain.util.UtilPreferences
import com.iamkatrechko.clipboardmanager.view.extension.onActivityResultOk

/**
 * Диалог выбора сортировки заметки
 * @author iamkatrechko
 *         Date: 01.11.2016
 */
class DialogSetOrderType : DialogFragment() {

    /**
     * Возвращает данные обратно во фрагмент
     * @param orderType тип сортировки
     */
    private fun sendResult(orderType: Int) {
        onActivityResultOk {
            putExtra("orderType", orderType)
        }
    }

    override fun onCreateDialog(bundle: Bundle?): Dialog {
        val itemsNames = OrderType.values().map { getString(it.nameResId) }.toTypedArray()
        val current = PrefsManager.getInstance().clipsOrderType

        return AlertDialog.Builder(context)
                .setTitle(R.string.title_sort)
                .setSingleChoiceItems(itemsNames, current.ordinal) { _, which ->
                    sendResult(which)
                    dismiss()
                }
                .create()
    }

    companion object {

        /**
         * Возвращает новый экземпляр диалога
         * @return новый экземпляр диалога
         */
        fun newInstance(): DialogSetOrderType {
            return DialogSetOrderType()
        }
    }
}
